### DISCUSSION TITLE
Has the allow_user_oidc_tokens been disabled?

### DISCUSSION THREAD

**Discussion by panta-123** [rucio/rucio#7108] [General]
I am looking through the code and find that the `from rucio.core.oidc import get_token_for_account_operation` has been replaced with `from rucio.core.oidc import request_token` . 
This seem to be an issue to use cilogon as token provider. The issue is that request_token operation uses client_credential flow which is not available in cilogon.

But we can use `get_token_for_account_operation` (i.e. via `__exchange_token_oidc`)  to get the access token using existing user token available in DB.

Any thoughts on this ?

**Comment by dchristidis** [comment]
The concept of user tokens being stored in the Rucio database and then being used for transfers will disappear entirely in the future. Please implement a function with the same signature as `request_token()` that uses the OAuth flow that CILogon supports. I’m afraid some additional work will may be required to read the original Rucio token from `idpsecrets.json`. It would be best to open an issue with the proposed modification prior to beginning the implementation.
